### PR TITLE
feat: isExtraReward mapping and tests

### PR DESCRIPTION
**Pull request by Hephyrius**

### CODE REVIEW

File: test/compounder/GenericUnionVault.spec.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import hre, { ethers } from "hardhat";
-import { expect } from "chai";
-import { Signer } from "ethers";
+import { expect, util } from "chai";
+import { Signer, utils } from "ethers";
```
Comment (Hephyrius) [inline test/compounder/GenericUnionVault.spec.ts:3]: Not sure how these imports snuck in

File: contracts/compounder/GenericVault.sol [hunk 2]
```diff
@@ -72,11 +73,12 @@ contract GenericUnionVault is ERC20, IERC4626, Ownable, ReentrancyGuard {
     /// @param _reward VirtualBalanceRewardPool address
     /// @return bool success
     function addExtraReward(address _reward) external onlyOwner notToZeroAddress(_reward) returns (bool) {
-        if (extraRewards.length >= 12) {
+        if (extraRewards.length >= 12 || isExtraReward[_reward]) {
```
Comment (phijfry) [inline contracts/compounder/GenericVault.sol:76]: TBF we can probably make this a require. It's `return false` on the BaseRewardPool where this is copied from as to not bork the pool but in this case a require revert makes more sense.

**Review by phijfry — APPROVED** [review]
Smol tweak but otherwise lgtm 🖖

File: contracts/compounder/GenericVault.sol (hunk 2 above)
Comment (Hephyrius) [reply contracts/compounder/GenericVault.sol]: Sure thing, just added the revert and revert test 👍

**Pull request merged**